### ISSUE TITLE
[Feature Request] Model Interpretability | Feature Importance

### ISSUE THREAD

**Issue by mafattma** [cornellius-gp/gpytorch#1197]
# 🚀 Feature Request

Model interpretability is very important and useful to convince non data scientists to trust the model. 
Several methods are available (LIME, SHAP, ...), however, libraries that implement such methods are focused on scikit-learn API. 
I found Captum, a library for PyTorch models, but it targets deep learning architectures.
Is it possible to include a feature importance method in GpyTorch?
Thank you!

**Comment by KeAWang** [comment]
The goal of gpytorch is to provide a flexible development framework like pytorch for building Gaussian process models. The user is then free to build anything they would like using gpytorch. Thus we do not aim to be like scikit-learn and will likely not natively support the features you mentioned within gpytorch.

**Comment by jacobrgardner** [comment]
@mafattma One way people often look at feature importance in the context of GPs is by using an ARD kernel, and then looking at the lengthscales. In general, dimensions with long lengthscales relative to the distances typically found in those dimensions correspond to fairly "unimportant" features in the sense that a long lengthscale means that you have to change that feature a lot to affect the output function value (e.g., label or target) at all.

If you want to use a NN + GP with toolbox feature extractors, then those should pretty much work out of the box with a GP last layer. As @KeAWang suggested, idk if GPyTorch is the right place to implement those.

**Comment by ginward** [comment]
> @mafattma One way people often look at feature importance in the context of GPs is by using an ARD kernel, and then looking at the lengthscales. In general, dimensions with long lengthscales relative to the distances typically found in those dimensions correspond to fairly "unimportant" features in the sense that a long lengthscale means that you have to change that feature a lot to affect the output function value (e.g., label or target) at all.
> 
> If you want to use a NN + GP with toolbox feature extractors, then those should pretty much work out of the box with a GP last layer. As @KeAWang suggested, idk if GPyTorch is the right place to implement those.

Thanks! This has been really helpful to me.
